system/Model.php

```diff
@@ -814,12 +818,11 @@ protected function objectToRawArray($data, bool $onlyChanged = true, bool $recur
 
         // Always grab the primary key otherwise updates will fail.
         if (
-            // @TODO Should use `$data instanceof Entity`.
+            // @TODO Should add an interface for this?
             method_exists($data, 'toRawArray')
             && (
                 ! empty($properties)
                 && ! empty($this->primaryKey)
-                && ! in_array($this->primaryKey, $properties, true)
```

Comment:
Why are we removing this check?

Reply:
This does not make sense.
`$this->primaryKey` is the column name like `id`.
`$properties` is an array with `property_name => value`.
If `id` is in the array values, the condition is `false`.

I thought it was a mistake, and it should use `array_key_exists()`, but
when the primary key value is binary data and it is cast to a string by Entity casting,
we need to set raw binary data even when there is a string primary key value in `$properties`.

Without casting, if there is a primary key value in `$properties`, we don't need to set it again,
but setting the same value is no harm.

Reply:
Oh.. yes, that makes sense. Thanks.